modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C

```diff
@@ -403,24 +403,36 @@ PolygonConcentricCircleMeshGeneratorBase::PolygonConcentricCircleMeshGeneratorBa
     paramError("outward_interface_boundary_names",
                "If provided, the length of this parameter must be identical to the total number of "
                "interfaces.");
-  if ((_has_rings || _background_intervals == 1) && _background_block_ids.size() > 1)
+  const unsigned int num_total_background_layers =
+      _background_intervals + _background_inner_boundary_layer_params.intervals +
+      _background_outer_boundary_layer_params.intervals;
+  if ((_has_rings || num_total_background_layers == 1) && _background_block_ids.size() > 1)
     paramError("background_block_ids",
                "This parameter must be either unset or have a unity length when ring_radii is "
-               "provided or background_intervals is unity.");
-  if ((_has_rings || _background_intervals == 1) && _background_block_names.size() > 1)
+               "provided or background intervals (including boundary layers) is unity.");
+  if ((_has_rings || num_total_background_layers == 1) && _background_block_names.size() > 1)
     paramError("background_block_names",
                "This parameter must be either unset or have a unity length when ring_radii is "
-               "provided or background_intervals is unity.");
+               "provided or background intervals (including boundary layers) is unity.");
+  if (!_has_rings && num_total_background_layers > 1 && _quad_center_elements &&
+      _background_block_ids.size() == 1)
+    _background_block_ids.insert(_background_block_ids.begin(), _background_block_ids.front());
   if ((!_has_rings && _background_intervals > 1) &&
       (!_background_block_ids.empty() && _background_block_ids.size() != 2))
     paramError("background_block_ids",
                "This parameter must be either unset or have a length of two when ring_radii is not "
-               "provided and background_intervals is not unity.");
+               "provided and background intervals (including boundary layers) is not unity. It can "
+               "optionally to have a unity length if `quad_center_elements` is enabled.");
+  if (!_has_rings && num_total_background_layers > 1 && _quad_center_elements &&
+      _background_block_names.size() == 1)
+    _background_block_names.insert(_background_block_names.begin(),
+                                   _background_block_names.front());
   if ((!_has_rings && _background_intervals > 1) &&
       (!_background_block_names.empty() && _background_block_names.size() != 2))
     paramError("background_block_names",
                "This parameter must be either unset or have a length of two when ring_radii is not "
-               "provided and background_intervals is not unity.");
+               "provided and background intervals (including boundary layers) is not unity. It can "
+               "optionally to have a unity length if `quad_center_elements` is enabled.");
```

Comment:
If the center elements are quad, the inner most background can have one single block ids. Otherwise, it has to have two different block ids. 
Currently, even if the center quad elements are used, users have to input the same block ids twice. After this update, users are allowed to only provide it once.

Reply:
but the center elements are not part of the background?